afs/media/js/views/components/workflows/project-report-workflow/download-project-files.js

```diff
@@ -0,0 +1,141 @@
+define([
+    'jquery',
+    'js-cookie',
+    'knockout',
+    'arches',
+    'viewmodels/alert-json',
+    'templates/views/components/workflows/project-report-workflow/download-project-files.htm',
+], function($, Cookies, ko, arches, JsonErrorAlertViewModel, downloadFilesTemplate) {
+    function viewModel(params) {
+        var self = this;
+
+        this.projectValue = params.projectId;
+
+        const observationGraphId = '615b11ee-c457-11e9-910c-a4d18cec433a';
+        const digitalResourcegGraphId = '707cbd78-ca7a-11e9-990b-a4d18cec433a';
+        const fileNodeId = '7c486328-d380-11e9-b88e-a4d18cec433a';
+        const fileStatementContentNodeId = 'ca227726-78ed-11ea-a33b-acde48001122';
+        this.relatedPhysicalThings = ko.observableArray();
+        this.relatedObservations = ko.observableArray();
+        this.message = ko.observable();
+
+        this.ready = ko.computed(() => {
+            return self.relatedObservations().find((observation) => {
+                return !!observation.relatedFiles().find(file => file.selected() == true );
+            });
+        });
+
+        this.formatSize = function(size) {
+            var bytes = size;
+            if(bytes == 0) return '0 Byte';
+            var k = 1024;
+            var dm = 2;
+            var sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+            var i = Math.floor(Math.log(bytes) / Math.log(k));
+            return '(' + '<strong>' + parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + '</strong>' + sizes[i] + ')';
+        };
+
+        this.fileTableConfig = {
+            columns: Array(4).fill(null)
+        };
+
+        this.expandAll = function(bool) {
+            self.relatedObservations().forEach((observation) => {
+                observation.expanded(bool);
+            });
+        };
+
+        this.selectAll = function(bool) {
+            self.relatedObservations().forEach((observation) => {
+                observation.relatedFiles().forEach(file => file.selected(bool));
+            });
+            self.expandAll(true);
+        };
+
+        this.getFilesFromObservation = async() => {
+            let projectObservations, observation, digitalResources;
+            await window.fetch(arches.urls.related_resources + self.projectValue  + "?paginate=false")
```

Comment:
If you're using an await in an async context, there is no need to chain a "then".  You can simply "await" it as well and remove the chaining to make this easier to read.  For instance, instead of saying `.then(response=>response.json())` you can simply say `const response = await response.json();`.  This will cut down on your nested functions.